### PR TITLE
Make assets of releases available also

### PR DESCRIPTION
**Pull request by kiwiroy**
Some projects add assets in addition to the tar/zip files that are automatically generated. These are frequently post `bootstrap` or `make dist` products.

This PR makes these files available in the list of candidates for download.

The new property `include_assets` accepts a Boolean or regular expression which will filter on name.

**edit**: arrgh 5.8

### CODE REVIEW

File: lib/Alien/Build/Plugin/Download/GitHub.pm [hunk 1]
```diff
@@ -123,12 +148,24 @@ sub init
...
+              my @results = ({
                 filename => $release->{tag_name},
                 url      => $release->{tarball_url},
                 defined $version ? (version  => $version) : (),
-              );
-              \%h;
+              });
+
+              if (my $include = $self->include_assets) {
+                my $filter = is_regexp($include) ? 1 : 0;
```
Comment (plicease) [inline lib/Alien/Build/Plugin/Download/GitHub.pm]: `is_regexp` doesn't seem available on Perl 5.8, as per CI

```suggestion
                my $filter = ref($include) eq 'Regexp' ? 1 : 0;
```

File: t/alien_build_plugin_download_github.t [hunk 2]
```diff
@@ -234,6 +234,61 @@ subtest 'basic load' => sub {
...
+
+      use alienfile;
+
+      probe sub { 'share' };
+
+      share {
+        plugin 'Download::GitHub' => (
+          github_user => 'Perl5-Alien',
+          github_repo => 'dontpanic',
+          include_assets => qr/^alien/,
```
Comment (plicease) [inline t/alien_build_plugin_download_github.t:274]: can you also please add a test with `include_assets = 1`?

File: lib/Alien/Build/Plugin/Download/GitHub.pm [hunk 3]
```diff
@@ -9,6 +9,7 @@ use JSON::PP qw( decode_json );
 use Alien::Build::Plugin;
 use Alien::Build::Plugin::Download::Negotiate;
 use Alien::Build::Plugin::Extract::Negotiate;
+use re 'is_regexp';
```
Comment (plicease) [inline lib/Alien/Build/Plugin/Download/GitHub.pm]: also remove this.

**Comment by plicease** [comment]
Looks good in general!  Can you point me to an example repo that uses this?  I'm not familiar with this feature.

File: t/alien_build_plugin_download_github.t [hunk 4]
```diff
@@ -234,6 +234,61 @@ subtest 'basic load' => sub {
...
+    local $mock_response{content} = encode_json([
+      {
+        tag_name => 'v1.01',
+        tarball_url => 'https://api.github.com/repos/Perl5-Alien/dontpanic/tarball/v1.01',
+        assets => [
+          {
+              url => 'https://api.github.com/repos/Perl5-Alien/dontpanic/releases/assets/123456',
+              name => 'alien-dontpanic-v1.01.tar.xz',
+              browser_download_url => 'https://github.com/repos/Perl5-Alien/dontpanic/releases/download/dontpanic-v1.01/alien-dontpanic-v1.01.tar.xz',
+          }
```
Comment (plicease) [inline t/alien_build_plugin_download_github.t]: add an asset that doesn't pass the regex.

**Comment by kiwiroy** [comment]
Here's one https://github.com/seqan/seqan3/releases

**Comment by plicease** [comment]
Thanks!

**Comment by kiwiroy** [comment]
Awesome.

**Pull request merged**